devito/ir/clusters/cluster.py

```diff
@@ -105,11 +106,11 @@ def rebuild(self, *args, **kwargs):
                 raise ValueError("`exprs` provided both as arg and kwarg")
             kwargs['exprs'] = args[0]
 
-        return Cluster(exprs=kwargs.get('exprs', self.exprs),
-                       ispace=kwargs.get('ispace', self.ispace),
-                       guards=kwargs.get('guards', self.guards),
-                       properties=kwargs.get('properties', self.properties),
-                       syncs=kwargs.get('syncs', self.syncs))
+        return self.__class__(exprs=kwargs.get('exprs', self.exprs),
```

Comment:
Why are you doing this?

Reply:
reconstruction of potential Cluster subclasses

Reply:
Should we ratehr have a `_rebuild` ?

Reply:
well that's how the method is called :)